docs/reference/ingest/processors/set-security-user.asciidoc

```diff
@@ -1,18 +1,23 @@
 [[ingest-node-set-security-user-processor]]
 === Set Security User Processor
-Sets user-related details (such as `username`,  `roles`, `email`, `full_name`
-and `metadata` ) from the current
+Sets user-related details (such as `username`,  `roles`, `email`, `full_name`,
+`metadata`, `api_key`, `realm` and `authentication_type`) from the current
 authenticated user to the current document by pre-processing the ingest.
+The `api_key` property is only applicable when the authentication is
+through API key. It is an object containing two fields, `id` and `name`.
+When using API key authentication, the `realm` property refers to the realm
+from which the API key is created. It is also an object with two fields,
+`name` and `type`.
```

Comment:
I think describing the `realm` property in the space of API Keys would lead some users to read it as only applying to API Keys. It would be better to say something like:
```
The `api_key` property is only applicable when the authentication is
through API key. It is an object containing two fields, `id` and `name`.

The `realm` property is also an object with two fields, `name` and `type`.
When using API key authentication, the `realm` property refers to the realm
from which the API key is created. 
```

That said, I think we ought to actually expand this whole section and give examples, etc.
I'd be happy to see that in a followup PR though.

Reply:
[Updated](https://github.com/elastic/elasticsearch/pull/51305/files#diff-5c7ce044d2cc26ab68eb77577355fd22R6-R10). I would also prefer to have the example as a separate PR. The example code may need to be added as part of doc tests which require some work. Should I create an issue for it?

Reply:
> Should I create an issue for it?

Yes please.

Reply:
Issue created: #51907